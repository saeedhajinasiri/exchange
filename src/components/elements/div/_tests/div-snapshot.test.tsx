import React from "react";
import {cleanup, render} from "test";
import Div from '../div'
import Text from '../../text'

afterEach(cleanup)

describe('Test Div UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Div>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with children prop', () => {
    const children = <Text color={"primary"}>children</Text>
    const {asFragment} = render(<Div>{children}</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<Div className={'test-class-name'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with item scope prop', () => {
    const {asFragment} = render(<Div itemScope={'test-item-scope'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with item type prop', () => {
    const {asFragment} = render(<Div itemType={'test-item-type'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with item prop prop', () => {
    const {asFragment} = render(<Div itemProp={'test-item-prop'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with style prop', () => {
    const {asFragment} = render(<Div style={{}}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with dangerously set inner html prop', () => {
    const testData = {"@test": "test-data",};
    const {asFragment} = render(<Div dangerouslySetInnerHTML={{__html: JSON.stringify(testData)}}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with role prop', () => {
    const {asFragment} = render(<Div role={'test-role'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with dir prop', () => {
    const {asFragment} = render(<Div dir={'test-dir'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with id prop', () => {
    const {asFragment} = render(<Div id={'test-id'}>children</Div>);
    expect(asFragment()).toMatchSnapshot();
  });
});
