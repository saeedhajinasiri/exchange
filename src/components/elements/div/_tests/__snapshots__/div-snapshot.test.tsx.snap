// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Div UI Default 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with children prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
  >
    <p
      class="text-m-sm sm:text-t-sm md:text-d-sm text-primary text-left font-regular"
    >
      children
    </p>
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with className prop 1`] = `
<DocumentFragment>
  <div
    class="flex test-class-name"
    data-testid="div-test-id"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with dangerously set inner html prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
  >
    {"@test":"test-data"}
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with dir prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    dir="test-dir"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with id prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    id="test-id"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with item prop prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    itemprop="test-item-prop"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with item scope prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    itemscope=""
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with item type prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    itemtype="test-item-type"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with role prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
    role="test-role"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Test Div UI Test with style prop 1`] = `
<DocumentFragment>
  <div
    class="flex undefined"
    data-testid="div-test-id"
  >
    children
  </div>
</DocumentFragment>
`;
